### PR TITLE
update version and toolchain for GLFW

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/d3ab8138f0ae16d03af5af1e13adbd89 for a full test report.

**Comment by verdurin** [comment]
Local build error:
```
/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/posix_time.c: In function _glfwPlatformInitTimer:
/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/posix_time.c:42:31: error: CLOCK_REALTIME undeclared (first use in this function)
   42 |     _glfw.timer.posix.clock = CLOCK_REALTIME;
      |                               ^~~~~~~~~~~~~~
/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/posix_time.c:42:31: note: each undeclared identifier is reported only once for each function it appears in
/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/posix_time.c:47:9: warning: implicit declaration of function clock_gettime [-Wimplicit-function-declaration]
   47 |     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
      |         ^~~~~~~~~~~~~
[ 32%] Building C object src/CMakeFiles/glfw.dir/xkb_unicode.c.o
cd /dev/shm/GLFW/3.4/GCCcore-12.2.0/easybuild_obj/src && /eb/maint/software/GCCcore/12.2.0/bin/gcc -D_DEFAULT_SOURCE -D_GLFW_WAYLAND -D_GLFW_X11 -I/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/include -I/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src -I/dev/shm/GLFW/3.4/GCCcore-12.2.0/easybuild_obj/src -I/eb/maint/software/X11/20221110-GCCcore-12.2.0/include -O2 -ftree-vectorize -march=native -fno-math-errno -O3 -DNDEBUG -fPIC -Wall -std=c99 -MD -MT src/CMakeFiles/glfw.dir/xkb_unicode.c.o -MF CMakeFiles/glfw.dir/xkb_unicode.c.o.d -o CMakeFiles/glfw.dir/xkb_unicode.c.o -c /dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/xkb_unicode.c
/dev/shm/GLFW/3.4/GCCcore-12.2.0/glfw-3.4/src/posix_time.c:47:23: error: CLOCK_MONOTONIC undeclared (first use in this function)
   47 |     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
      |                       ^~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/glfw.dir/posix_time.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 14 out of 14 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/8036c70e7cf298efc506a04452021561 for a full test report.

File: easybuild/easyconfigs/g/GLFW/GLFW-3.4-GCCcore-12.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# build both static and shared libraries
+configopts = [
+    '-DBUILD_SHARED_LIBS=OFF',
+    '-DBUILD_SHARED_LIBS=ON',
+]
+
+
+sanity_check_paths = {
+    'files': ['include/%(name)s/glfw3.h', 'include/%(name)s/glfw3native.h'] +
+             ['lib/libglfw3.a', 'lib/libglfw.%s' % SHLIB_EXT],
```
Comment (Micket) [inline easybuild/easyconfigs/g/GLFW/GLFW-3.4-GCCcore-12.2.0.eb:35]:
```suggestion
             ['lib/libglfw3.a', 'lib/libglfw.%s' % SHLIB_EXT, 'lib/pkgconfig/glfw3.pc'],
```

I see no reason to remove the check for glfw3.pc. It's still present, and important.

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/9b601f6a35c8dbb45d3c986c42705a51 for a full test report.

**Review by fizwit — COMMENTED** [review]
lgtm

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Review by Micket — APPROVED** [review]
LGTM
I don't know what might be causing the normal time.h to break on centos7, but i don't think there is anything to do with this particular easyconfig and version. Maybe the toolchain itself is broken on centos7.
It's as identical as the previous toolchain version was, and this looks all very very standard. I will leave it to those on centos7 who are able to reproduce that issue to fix that.

**Comment by Micket** [comment]
Going in, thanks @fizwit!

**Pull request merged**